builder/pipeline.cc

```diff
@@ -659,6 +826,8 @@ stmt build_pipeline(node_context& ctx, const std::vector<buffer_expr_ptr>& input
 
   if (options.no_checks) {
     result = recursive_mutate<check>(result, [](const check* op) { return stmt(); });
+    // Simplify because some of the bodies might become empty after removing asserts.
+    result = simplify(result);
```

Comment:
I think we move the `simplify` + `fix_buffer_races` above to after removing the checks instead of adding another call to `simplify`